streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -436,6 +430,8 @@ private StateConsumer initialize() {
         } catch (final Exception fatalException) {
             closeStateConsumer(stateConsumer, false);
             startupException = new StreamsException("Exception caught during initialization of GlobalStreamThread", fatalException);
+        } finally {
+            initializationLatch.countDown();
```

Comment:
I think to keep the existing semantics, another one of these is needed in `shutdown`.

Reply:
I don't think we need to right now, given that `start()` blocks and thus `shutdown()` cannot be triggered as long as `init()` did not finish (cf https://issues.apache.org/jira/browse/KAFKA-7380)

But maybe good to add to be future proof...

Reply:
Interesting... 💭 I have added the countdown to the `shutdown()`.